File: backend/app/schemas/company.py
from pydantic import BaseModel


class CompanyHeaderDTO(BaseModel):
    ticker: str
    companyName: str
    industry: str
    marketCap: str
    price: float
    changePercent: float


class CompanyNewsDTO(BaseModel):
    title: str
    publishedAt: str
    source: str


class CompanyAISummaryDTO(BaseModel):
    summary: str
    bullCase: list[str]
    risk: list[str]
    watchItems: list[str]


class CompanyDataDTO(BaseModel):
    header: CompanyHeaderDTO
    news: list[CompanyNewsDTO]
    aiSummary: CompanyAISummaryDTO


class CompanyResponseDTO(BaseModel):
    success: bool = True
    data: CompanyDataDTO

File: frontend/src/types/company.ts
export interface CompanyHeaderDTO {
  ticker: string;
  companyName: string;
  industry: string;
  marketCap: string;
  price: number;
  changePercent: number;
}

export interface CompanyNewsDTO {
  title: string;
  publishedAt: string;
  source: string;
}

export interface CompanyAISummaryDTO {
  summary: string;
  bullCase: string[];
  risk: string[];
  watchItems: string[];
}

export interface CompanyDTO {
  header: CompanyHeaderDTO;
  news: CompanyNewsDTO[];
  aiSummary: CompanyAISummaryDTO;
}
